microprofile/server/src/main/java/io/helidon/microprofile/server/ServerCdiExtension.java

```diff
@@ -108,10 +109,34 @@ private void startServer(@Observes @Priority(PLATFORM_AFTER + 100) @Initialized(
 
         // make sure all configuration is in place
         if (null == jaxRsExecutorService) {
-            jaxRsExecutorService = ServerThreadPoolSupplier.builder()
+            Config serverConfig = config.get("server");
+            final java.lang.reflect.Method m;
+            if (serverConfig.get("virtual-threads").asBoolean().isPresent()) {
```

Comment:
`asBoolean().orElse(false)` - what if `virtual-threads: false` - you would execute as well

Reply:
Yep; just a dumb typo in this draft PR